src/tools/fuzzing/fuzzing.cpp

```diff
@@ -23,6 +23,22 @@ namespace wasm {
 
 namespace {
 
+// Construct a fresh signature type for a given signature, and return its heap
+// type. This is useful when we do not want to get the default behavior of
+// reusing the "canonical" heap type for a signature, which can cause problems
+// in closed-world mode - in that mode the canonical heap type may be private,
+// so we should use a fresh type whenever we construct a function that will be
+// exported.
+static HeapType getFreshSignatureHeapType(Signature sig) {
+  TypeBuilder builder(1);
+  builder.setHeapType(0, sig);
+  return (*builder.build())[0];
```

Comment:
This will usually not create a fresh type due to canonicalization, although it may happen to create a new type the first time it is called.

Reply:
Ah, the error happened in nominal mode, so I guess I just checked there. It did work in that case.

I guess for the isorecursive case we need to do... something more? But as we fuzz only nominal for now, this at least unblocks us right now.